### PR TITLE
Fixed the bug

### PR DESCRIPTION
**Pull request by cody-constine-ttd**

### CODE REVIEW

File: src/test/java/com/uid2/admin/vertx/SharingServiceTest.java [hunk 1]
```diff
@@ -593,6 +593,38 @@ void KeysetSetBadWhitelistSiteId(Vertx vertx, VertxTestContext testContext) {
...
+            put(3, new Keyset(3, 7, "test", Set.of(12, 25), Instant.now().getEpochSecond(),true, true));
+            put(4, new Keyset(4, 4, "test", Set.of(5), Instant.now().getEpochSecond(),true, true));
+        }};
+
+        setKeysets(keysets);
+
+        String body = "  {\n" +
+                "    \"allowed_sites\": [\n" +
+                "      22,\n" +
+                "      25,\n" +
```
Comment (jon8787) [inline src/test/java/com/uid2/admin/vertx/SharingServiceTest.java:612]: should we add a site that doesn't exist? That was the scenario that we're fixing

File: src/main/java/com/uid2/admin/vertx/service/SharingService.java [hunk 2]
```diff
@@ -134,13 +136,18 @@ private void handleSetKeyset(RoutingContext rc) {
...
             }
 
 
             final Set<Integer> newlist;
 
             if (allowedSites != null){
-                OptionalInt firstInvalidSite = allowedSites.stream().mapToInt(s -> (Integer) s).filter(s -> siteProvider.getSite(s) == null).findFirst();
+                final Set<Integer> existingAllowedSites = existingSites;
+                OptionalInt firstInvalidSite = allowedSites.stream().mapToInt(s -> (Integer) s).filter(s -> {
+                    if(existingAllowedSites.contains(s)) return false;
```
Comment (jon8787) [inline src/main/java/com/uid2/admin/vertx/service/SharingService.java:148]: would something like `.filter(s -> !existingAllowedSites.contains(s) && !isSiteIdEditable(s))` work

**Review by jon8787 — APPROVED** [review]

File: src/test/java/com/uid2/admin/vertx/SharingServiceTest.java (hunk 1 above)
Comment (cody-constine-ttd) [reply src/test/java/com/uid2/admin/vertx/SharingServiceTest.java:612]: Yes this is what it is doing, 25 does not exist

Comment (cody-constine-ttd) [reply src/test/java/com/uid2/admin/vertx/SharingServiceTest.java:615]: Added some comments

**Pull request merged**